### ISSUE TITLE
Graphs plugin op overlap

### ISSUE THREAD

**Issue by matthewkotila** [tensorflow/tensorboard#3878]
## Environment information (required)

no personal environment--i'm using the tensorboard from this google colab for simplicity:
* https://colab.research.google.com/github/tensorflow/tensorboard/blob/master/docs/graphs.ipynb

## Issue description

bug description:
* 'detached' graphs plugin ops (auxiliary nodes?) overlap with each other obscuring view/information potentially useful to user
<img width="701" alt="image" src="https://user-images.githubusercontent.com/7692737/88324969-512b8e80-ccd9-11ea-981c-d8a7492b46a3.png">

reproduction instructions:
1. go to [the google colab mentioned above](https://colab.research.google.com/github/tensorflow/tensorboard/blob/master/docs/graphs.ipynb) (used this so i could upload my own graphdef pbtxt to view)
2. click the `Run cell` button in the `Op-level graph` section of the colab
3. within tensorboard, navigate to `GRAPHS` tab/plugin
4. click the `Choose File` button on left
5. upload the unzipped [graphdef.pbtxt.zip](https://github.com/tensorflow/tensorboard/files/4968123/graphdef.pbtxt.zip)
6. notice how some 'detached' ops on right side overlap with each other

## Question(s)

1. is this an issue with the particular graphdef pbtxt i've used? (i.e. is it misformatted?)

**Comment by bileschi** [comment]
Thank you for the report @matthewkotila .  I can reproduce this.  This behavior is clearly not by design, but an artifact of the Graph viewer being designed for an age when graphs were siginificantly smaller than they are in 2020.  It is probably not an issue with your graphdef.  If you click "add to main graph", you can see where the resnet50 block fits within the full structure.

As you can see, the graph viewer does not know how to handle the repeated loss elements in a nice, abstract way:

![image](https://user-images.githubusercontent.com/547150/88699309-2c6a5900-d0d5-11ea-9f4d-e5909390568d.png)

**Comment by matthewkotila** [comment]
hi @bileschi, thanks for the quick response 🙏

i see you mentioned a user-side workaround--is there a developer-side fix (intended to be) in the pipeline?

**Comment by bileschi** [comment]
Not really.  We love the graph plugin and want to invest in a number of improvements, but right now are swamped by a few other high priority items, such as the Polymer 3 migration.  There are a number of things we want to do to make the graph plugin excellent, but don't have the extra engineering capacity to make those investments:

*  Saving personalized layouts https://github.com/tensorflow/tensorboard/issues/294
*  Make it easier to import (upload) and export (download) graphs of different types
*  Improve collapsing of repeated patterns https://github.com/tensorflow/tensorboard/issues/572
*  Improve support of higher-level graph abstractions (like Keras)
*  Improve layout logic in general (https://github.com/tensorflow/tensorboard/issues/1791)

And many more at tag `plugin:graph`

Contributions welcome!

**Comment by matthewkotila** [comment]
understandable. thanks for the follow-up--wishing y'all the best 😄🙏
